src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIError.cs

```diff
@@ -30,12 +33,12 @@ public SNIError(SNIProviders provider, uint nativeError, uint sniErrorCode, stri
             this.exception = null;
         }
 
-        public SNIError(SNIProviders provider, uint sniErrorCode, Exception sniException)
+        public SNIError(SNIProviders provider, uint sniErrorCode, Exception sniException, uint nativeErrorCode = 0)
         {
             this.lineNumber = 0;
             this.function = string.Empty;
             this.provider = provider;
-            this.nativeError = 0;
+            this.nativeError = nativeErrorCode;
```

Comment:
NIT: `this` keyword is not necessary as the variable name is not same as the fieldname. In fact the only part that needs `this` keyword is `provider` which field's name and argument's name is exactly the same.